src/bin/redact.js

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env babel-node
+
+const parser = require('../cdoFlavoredParser');
+const parseArgs = require('minimist')
+const fs = require('fs');
+
+const argv = parseArgs(process.argv.slice(2));
+
+const inputfile = argv._[0];
+let inputdata = fs.readFileSync(inputfile);
+try {
+  inputdata = JSON.parse(inputdata);
+} catch (e) {
```

Comment:
nit: prefer to just catch SyntaxError here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse#Exceptions

Reply:
I'm not sure I agree; conditional catch clauses in JS exist but they are not a standard, so catching just a SyntaxError would mean doing something like

```javascript
try {
    inputdata = JSON.parse(inputdata);
} catch (e) {
    if (e instanceof SyntaxError) {
        inputdata = inputdata.toString();
    } else {
        throw(e)
    }
}
```

which just seems silly, particularly given that by the documentation you linked, SyntaxError is the _only_ thing we expect to see thrown.

Do you prefer to catch just a SyntaxError based on an existing standard, or is it personal preference?

Reply:
Good points Elijah, what you have in the pull request looks fine -- I must be thinking in another language. Sorry to make you explain all that!

Reply:
Thanks for your thoroughness!